pkg/cli/cliexecuter.go

```diff
@@ -44,7 +44,7 @@ func NewExecuter(config *restclient.Config) (Executer, error) {
 }
 
 // Execute builds and runs a Post request on contianer for SPDY (shell) executor
-func (e executer) Execute(pod *corev1.Pod, command []string) (stdout string, stderr string, err error) {
+func (e executer) Execute(pod *corev1.Pod, nodeParams []string, command []string) (stdout string, stderr string, err error) {
```

Comment:
Is the `nodeParams` parameter included in the function solely for debugging purposes, right?
I think it is better to avoid using arguments that the function doesn't actually require. Instead, you could consider logging the error along with the `nodeParams` content in the caller function.
WDYT?

Reply:
> Is the nodeParams parameter included in the function solely for debugging purposes, right?

Correct, I do want that parameter for differentiating between log outputs of different FAR CRs (and most importantly success log output that I can look for in FAR e2e tests, see function **[buildExpectedLogOutput](https://github.com/medik8s/fence-agents-remediation/pull/20/files#diff-b9729ac4458c7637438e3bf558999b35adedcaf4d13af1e3eb752db501fd4997R266-R277)**).

Reply:
> I think it is better to avoid using arguments that the function doesn't actually require.

👍🏻 

> Instead, you could consider logging the error along with the nodeParams content in the caller function.

The caller function is [`reconcile`](https://github.com/medik8s/fence-agents-remediation/pull/55/files#diff-05147ee3f0d225b5a60754b8b9bff2c569deb0a9d7eaddae33af680291d493a1R116), so IIUC you are suggesting to  log _nodeParams_  in reconcile after we call [execute](https://github.com/medik8s/fence-agents-remediation/pull/55/files#diff-05147ee3f0d225b5a60754b8b9bff2c569deb0a9d7eaddae33af680291d493a1R116)? Sounds better, but I believe we need to do more changes for that. ATM in reconcile we print on failure result or on an error, and I would like to have the nodeParams on success situation as well.

Therefore I see three options (could be more):

1. Don't log in `pkg/cli/cliexecuter.go`-  just log on caller function. This why we don't share the `nodeParams` parameter, and we don't log twice.
2. Don't share the `nodeParams` parameter to `pkg/cli/cliexecuter.go`, and we use the parameter for logging in reconcile. We end up with two logs stating the same thing, just in a different manner.
3. Keep it as it is now, and don't add more changes. One log output about CLI execution, but with the minimal cost of sharing a parameter.

Reply:
Thinking more about it. Would it make more sense to not modify _buildFenceAgentParams_ function, and share the node's name rather than `nodeParams`?. Both of them are node identifiers....

Reply:
Since the idea of adding this parameter is for differentiating between log outputs of different FAR CRs and ATM there is only one E2E test, so I will dismiss this change, and handle it in the future.

Reply:
Sorry for replying so late :facepalm: 

> Thinking more about it. Would it make more sense to not modify buildFenceAgentParams function, and share the node's name rather than nodeParams?. Both of them are node identifiers

Do you only want to log a Node identifier? If that so, is it enough the new log just above calling `.Executer`?

Reply:
I understand. In any case, since the [log before calling `Executer.Execute`](https://github.com/medik8s/fence-agents-remediation/pull/55/files#diff-f2169f37331a57117047ad17f0b9bac545f2606f32eadce37a69fedb2b92cc61) has everything `Executer.Execute` has (except `stdout`), consider logging there.

Reply:
But what value will it add? Since I want to easily search (by reviewing in log and in E2E) for a substring from log that has stdout and nodeIdentifier together.

Reply:
you could log *after*, when you also have `stdout`

Reply:
> you could log after, when you also have stdout

Sure, I am aware of that. I just don't know which option is better from above. I only understand that 3 is not recommended.

Reply:
I'd say option 1 looks good